pydantic/types.py

```diff
@@ -2425,7 +2425,7 @@ class Model(BaseModel):
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
         return core_schema.with_info_after_validator_function(
             function=self.decode_str,
-            schema=super(EncodedStr, self).__get_pydantic_core_schema__(source=source, handler=handler),  # noqa: UP008
+            schema=super().__get_pydantic_core_schema__(source=source, handler=handler),  # noqa: UP008
```

Comment:
The noqa comment was actually there to prevent the `super()` change introduced. I'm not sure if it was intentional?

Reply:
So just do not filter that file only in corrected PR? 🤔

Reply:
Confirmed from this comment: https://github.com/pydantic/pydantic/pull/7396#issuecomment-1727348592

This one should be left as is.

So basically only the changes in `tests/benchmarks` can be kept :)

Reply:
@kloczek,

Go for it. Looks like there are a few linting issues as well...

Reply:
> OK I've rediffed that to modify only tests/ but I see pydantic/types.py can be updated as well.

No as I said the explicit `super` call is required as mentioned in the following comment: https://github.com/pydantic/pydantic/pull/7396#issuecomment-1727348592

Reply:
So @kloczek,

Just the benchmarks changes should do